oslc4j-client/src/main/java/org/eclipse/lyo/oslc4j/client/OslcClient.java

```diff
@@ -441,6 +441,10 @@ public String lookupServiceProviderUrl(final String catalogUrl, final String ser
 	{
 		String retval = null;
 		Response response = getResource(catalogUrl,OSLCConstants.CT_RDF);
+		if (response.getStatus() != HttpStatus.SC_OK) {
+			logger.error("Cannot read "+catalogUrl+" status: "+response.getStatus());
+			return null;
```

Comment:
Why return null?

Reply:
changed to return retval (which will be null in this case)